types/nodemailer-mailgun-transport/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for nodemailer-mailgun-transport 1.3
+// Project: https://github.com/orliesaurus/nodemailer-mailgun-transport
+// Definitions by: Oto Ciulis <https://github.com/otociulis/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as nodemailer from 'nodemailer';
+
+declare namespace mailgunTransport {
+    interface AuthOptions {
+        api_key: string;
+        domain?: string;
+    }
+}
+
+declare function mailgunTransport(options: mailgunTransport.AuthOptions): nodemailer.Transport;
```

Comment:
I was checking other definitions of nodemailer plugins and they all return Transport which is input parameter of nodemailer.createTransport method (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/index.d.ts#L77) and this function then returns type Transporter. Not sure why Transport.close() is marked as mandatory method because it looks like it's needed only for SMTP-transport; SES and JSON transports (https://github.com/nodemailer/nodemailer/blob/master/lib/ses-transport/index.js, https://github.com/nodemailer/nodemailer/blob/master/lib/json-transport/index.js) doesn't have them either.